exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SystemOptionManager.java

```diff
@@ -48,11 +50,46 @@
  * Only one instance of this class exists per drillbit. Options set at the system level affect the entire system and
  * persist between restarts.
  */
+
+/**
+ *  Drill has two different config systems each with its own namespace.First being the HOCON based boot time config
+ *  system.This is a hierarchical system where the top layers override the bottom ones in the following order
+ *
+ *  Java System Options
+ *  distrib.conf
+ *  drill-override.conf
+ *  drill-module.conf
+ *
+ *  These are the options that are set before the drill starts.But once drill starts System or session options can be
+ *  modified using ALTER SYSTEM/SESSION.Even this system provides inheritance sytle in the following order
+
+ *  Session options
+ *  System options
+ *  Hardcoded defaults
+
+ *  But system/session options have a validator and the validator has a hard coded default value for every option. In
```

Comment:
This description would be great as a comment for the pull request. Now, image that this code is committed to Drill, and someone reads this a year from now. By then, the comment will be describing ancient history.

We can see that this kind of comment should describe the system the way it is (or will be) after this change without need to reference history. For those who might be familiar with the old way, we often point them to the JIRA that describes the change.

Example:

"Default for system properties are externalized to the boot-time config file. See DRILL-5547. ..."